### DISCUSSION TITLE
How to disable SSL verification for proxied host?

### DISCUSSION THREAD

**Discussion by vggscqq** [NginxProxyManager/nginx-proxy-manager#3332] [Q&A] [answered]
Previously I was using simple nginx config for reverse proxy my services, all of them have self-signed SSL cert. Self-signed certs are not trusted by nginx reverse proxy server thus I had to disable cert verification like that

```
location / {
    proxy_pass https://10.10.0.1:1234/;
    proxy_ssl_verify off;
}
```

I tried to use following configuration in "Advanced" tab:

```
location / {
    proxy_ssl_verify off;
}
```

and

```
location / {
    proxy_pass https://10.10.0.1:1234/;
    proxy_ssl_verify off;
}

```

Using first one NPM instead of being reverse proxy tries to open local folder and using second one renders host Offline in GUI.

My idea is to use NPM same way as I used nginx:

`Internet <-Let's encrypt cert-> NPM server <-self-signed cert-> service server.`

**Comment by pwuersch** [comment]
Have you found a solution to this use-case yet? I‘m looking for something similar in my setup.

**Reply by vggscqq** [reply, marked as answer]
Hi, yes I did.
Go to your proxy host custom locations and define `/` (root) location.
Scheme should be https.
Under advanced settings use this

```
location / {
    proxy_pass https://self-signed.badssl.com/:443;
    proxy_ssl_verify       off;
    proxy_ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
}
```

**Reply by donniewr** [reply]
> Hi, yes I did. Go to your proxy host custom locations and define `/` (root) location. Scheme should be https. Under advanced settings use this
> 
> ```
> location / {
>     proxy_pass https://self-signed.badssl.com/:443;
>     proxy_ssl_verify       off;
>     proxy_ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
> }
> ```

I'm using CloudPanel and I also need to disable TLS verification for a self-signed certificate, but this code doesn't work for me. (bad gateway)

**Reply by barbar-binks** [reply]
May I ask how you ended up solving this, your issue is the one that resembles mine the most. I am running Ngnix Proxy Manager on my network and I want to redirect my domain to a docker container running Nextcloud on Unraid. I want to ignore the self-signed certificate from the container, in your example https://self-signed.badssl.com/:443 is that a placeholder for your actual domain? you answer also renders my host offline.

**Reply by donniewr** [reply]
I managed to make it work using this configuration, but your issue might be different.

Add this in advanced tab.

location / {
    proxy_pass $forward_scheme://$server:$port;
    proxy_set_header Host test.com;
    proxy_ssl_name test.com;
    proxy_ssl_server_name on;
    proxy_redirect $forward_scheme://$server:$port /;
}

**Reply by vggscqq** [reply]
> is that a placeholder for your actual domain?

Yes, actually I'm using ip addresses with self-signed certs and with proxy I can get Let's Encrypt cert.
So it's working like that:
```
intenet --> [ LE Cert ] --> npm --> [Self-Signed cert] --> my_service
```

```
location / {
    proxy_pass https://10.13.0.53:9856;
    proxy_ssl_verify       off;
    proxy_ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
}
```

**Reply by IsaakBenif** [reply]
> > is that a placeholder for your actual domain?
> 
> Yes, actually I'm using ip addresses with self-signed certs and with proxy I can get Let's Encrypt cert. So it's working like that:
> 
> ```
> intenet --> [ LE Cert ] --> npm --> [Self-Signed cert] --> my_service
> ```
> 
> ```
> location / {
>     proxy_pass https://10.13.0.53:9856;
>     proxy_ssl_verify       off;
>     proxy_ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
> }
> ```
It looks like it stopped working in version 2.12.3. NPM verifies the validity of the certificate even with proxy_ssl_verify off; the error appears in the logs "[crit] 743#743: *4921 ssl_d_handshake() failed (SSL: error:0A000458:SSL routines::tlsv1 unrecognized name:SSL alert number 112) while SSL handshaking to upstream".
I think there is a conflict with other configuration files. 
It would be nice for the author to add the "proxy ssl verify off" checkbox. This is a popular NPM usage scenario when we want to encrypt the transmitted data with a self-signed certificate.

**Reply by vggscqq** [reply]
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/4492

**Reply by IsaakBenif** [reply]
I managed to solve the problem. The algorithm is as follows:
Proxy Host -> ... -> Edit
Custom locations -> Add location
location: "/" Scheme: "https" Forward Hostname / IP: "backend-IP" Forward Port: "Backend-Port"

click on the gear button and add:

```
proxy_ssl_server_name on;
proxy_ssl_verify off;
proxy_ssl_name test.com;
```
It's working! > [Image](https://github.com/user-attachments/assets/8b1785d0-0364-4b15-8db5-106bcf8c72d6)

**Reply by chrizbee** [reply]
> I managed to solve the problem. The algorithm is as follows: Proxy Host -> ... -> Edit Custom locations -> Add location location: "/" Scheme: "https" Forward Hostname / IP: "backend-IP" Forward Port: "Backend-Port"
> 
> click on the gear button and add:
> 
> ```
> proxy_ssl_server_name on;
> proxy_ssl_verify off;
> proxy_ssl_name test.com;
> ```
> 
> It's working! > [Image](https://github.com/user-attachments/assets/8b1785d0-0364-4b15-8db5-106bcf8c72d6)

Is it still working for you? I used your configuration and got "502 Bad Gateway"...

**Reply by IsaakBenif** [reply]
> > I managed to solve the problem. The algorithm is as follows: Proxy Host -> ... -> Edit Custom locations -> Add location location: "/" Scheme: "https" Forward Hostname / IP: "backend-IP" Forward Port: "Backend-Port"
> > click on the gear button and add:
> > ```
> > proxy_ssl_server_name on;
> > proxy_ssl_verify off;
> > proxy_ssl_name test.com;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It's working! > [Image](https://github.com/user-attachments/assets/8b1785d0-0364-4b15-8db5-106bcf8c72d6)
> 
> Is it still working for you? I used your configuration and got "502 Bad Gateway"...

Yes, it works for me. Perhaps it has a server type at the end of the path? I use Ngnix on the backend.

**Reply by CT-9901** [reply]
> I managed to solve the problem. The algorithm is as follows: Proxy Host -> ... -> Edit Custom locations -> Add location location: "/" Scheme: "https" Forward Hostname / IP: "backend-IP" Forward Port: "Backend-Port"
> 
> click on the gear button and add:
> 
> ```
> proxy_ssl_server_name on;
> proxy_ssl_verify off;
> proxy_ssl_name test.com;
> ```
> 
> It's working! > [Image](https://github.com/user-attachments/assets/8b1785d0-0364-4b15-8db5-106bcf8c72d6)

I would like to do do same thing. But I am a little rusty on the vocab. So what is your "backend-IP"? The internal ip of the server I want the proxy host to lead to or the ip of the server that hosts the nginx container? Same thing for the backend-port: Do I need to fill in the port of the server I want the proxy host pointing at or some port on the server hosting nginx?

Right now I am going with the first option and I get an internal error when trying to request an ssl cert for my server.

**Reply by CT-9901** [reply]
> > I managed to solve the problem. The algorithm is as follows: Proxy Host -> ... -> Edit Custom locations -> Add location location: "/" Scheme: "https" Forward Hostname / IP: "backend-IP" Forward Port: "Backend-Port"
> > click on the gear button and add:
> > ```
> > proxy_ssl_server_name on;
> > proxy_ssl_verify off;
> > proxy_ssl_name test.com;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It's working! > [Image](https://github.com/user-attachments/assets/8b1785d0-0364-4b15-8db5-106bcf8c72d6)
> 
> I would like to do do same thing. But I am a little rusty on the vocab. So what is your "backend-IP"? The internal ip of the server I want the proxy host to lead to or the ip of the server that hosts the nginx container? Same thing for the backend-port: Do I need to fill in the port of the server I want the proxy host pointing at or some port on the server hosting nginx?
> 
> Right now I am going with the first option and I get an internal error when trying to request an ssl cert for my server.

Answering to my own question.
For me the entry under "Advanced" worked out with:

`location / {
    proxy_pass https://192.168.178.203:8123;
    proxy_ssl_verify off;
    proxy_ssl_name test.com;
}`

Since I am trying to connect a home assistant instance I had to add the ip of the nginx host server under "trusted_proxies" to make the nginx request come through to the home assistant server. Took me some time to figure that out
